velox/functions/prestosql/benchmarks/ArraySumBenchmark.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include "velox/functions/Macros.h"
+#include "velox/functions/Registerer.h"
+#include "velox/functions/lib/benchmarks/FunctionBenchmarkBase.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+
+namespace {
+
+template <typename T>
+struct udf_array_sum {
```

Comment:
out of curiosity: how do these two versions compare? Ideally we would like to move functions to use the simple framework as much as can. 

Cc: @laithsakka

Reply:
@pedroerp Currently only one version works. The simple version declared in ArraySumBenchmark.cpp has build errors related to it. I will post a comparison as soon as I get it working.

Reply:
Great to know the simple function version is fixed and performs better. Do we need to use the vector version at all in the benchmark comparison? I thought the benchmarks compare the across commits to ensure there is no regression.

Reply:
@majetideepak All the benchmarks I have looked at had a different implementation to benchmark against so that's what I did.

Reply:
@jwyles-ahana that makes sense. It's expected that the vectorized implementation would be sort of a "lower-bound" if compared to simple functions, as simple function will essentially in the end of the day translate to a vectorized function. 

The important part would be to understand any gaps between the simple and vectorized profiles, and make ensure we're working towards making simple functions as efficient as their vectorized's counterparts. 

Thanks for the analysis!